### ISSUE TITLE
Type queries in array types can still cause 'undefined symbol' errors

### ISSUE THREAD

**Issue by dlongnecke-cray** [chapel-lang/chapel#23882]
There are still type queries embedded in array types that are not working. While primitive bit-width queries (e.g., `[...] int(?w)` work now after #23807, an arguably more general case still causes "undefined symbol" errors.

```chapel
class C { type T; }

var a: [0..1] owned C(int)?;

proc foo(x: [] owned C(?w)?) { writeln(w:string); }

foo(a);
```

Results in an undefined symbol error for `w`.

Rather than represent just this one specific case, I'd like for this to be a blanket issue requesting support for type queries embedded in array types more broadly.

**Comment by DanilaFe** [comment]
Duplicate of https://github.com/chapel-lang/chapel/issues/13262. I'll be adding a future separately.
